### PR TITLE
Move the tests from EC to RC for 4.14

### PR DESCRIPTION
**Pull request by liangxia**
We have [stable 4.14](https://amd64.ocp.releases.ci.openshift.org/releasestream/4-stable/release/4.14.0-rc.0) on Sep 7, 2023

### CODE REVIEW

**Comment by jianlinliu** [comment]
Shall we update `releases.latest` filed to switch over `4-dev-preview` ?

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-stable-azure-stack-ipi-proxy-fips-f14

**Comment by liangxia** [comment]
Verified it's using the expected payload,
```
INFO[2023-09-08T03:09:49Z] Resolved release latest to quay.io/openshift-release-dev/ocp-release:4.14.0-rc.0-x86_64 
```

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-stable.yaml [hunk 1]
```diff
@@ -59,15 +59,13 @@ releases:
       product: ocp
       version_bounds:
         lower: 4.14.0-0
-        stream: 4-dev-preview
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-stable.yaml:62]: Shall we update it to use `channel: candidate` just like what we did in release-4.13__{arm64,amd64}-stable.yaml?

Comment (liangxia) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-stable.yaml:62]: Do we have that already ?

**Comment by liangxia** [comment]
Replaced by https://github.com/openshift/release/pull/43134

/close

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-stable.yaml (hunk 1 above)
Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__amd64-stable.yaml:62]: In public stream, I can see it:
```
[root@preserve-jialiu ~]# curl --silent --header 'Accept:application/json' 'https://api.openshift.com/api/upgrades_info/v1/graph?channel=candidate-4.14' | jq '.nodes[] |select(.version == "4.14.0-rc.0")'
{
  "version": "4.14.0-rc.0",
  "payload": "quay.io/openshift-release-dev/ocp-release@sha256:1d2cc38cbd94c532dc822ff793f46b23a93b76b400f7d92b13c1e1da042c88fe",
  "metadata": {
    "io.openshift.upgrades.graph.release.channels": "candidate-4.14",
    "io.openshift.upgrades.graph.release.manifestref": "sha256:1d2cc38cbd94c532dc822ff793f46b23a93b76b400f7d92b13c1e1da042c88fe"
  }
}
```

**Pull request closed without merging**